presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -167,10 +167,11 @@ private static ParquetPageSource createParquetPageSource(
         try {
             FileSystem fileSystem = hdfsEnvironment.getFileSystem(user, path, configuration);
             FSDataInputStream inputStream = hdfsEnvironment.doAs(user, () -> fileSystem.open(path));
-            ParquetMetadata parquetMetadata = MetadataReader.readFooter(inputStream, path, fileSize);
+            dataSource = new HdfsParquetDataSource(new ParquetDataSourceId(path.toString()), estimatedFileSize, inputStream, stats, options);
```

Comment:
This needs to handle start + length > discovered fileSize and start >= discovered file size as well.

Reply:
I believe the split region (i.e., `start` and `length`) is handled below when `footerBlocks`.  Basically, the code filters the block as declared in the footer to only those that start withing the split region, and only those blocks are processed.  If the split region is off the end of the file it shouldn't matter because no actual blocks can appear in that region.